mutt/string.c

```diff
@@ -959,3 +959,71 @@ const char *mutt_str_getenv(const char *name)
 
   return NULL;
 }
+
+/**
+ * Remove all occurrences of substring, ignoring case.
+ * @param str     String containing the substring
+ * @param target  Target substring for removal
+ * @retval 0      String contained substring and substring was removed successfully.
+ * @retval != 0   String did not contain substring.
+ */
+int mutt_str_remall_strcasestr(char *str, const char *target)
+{
+  // There's no point in removing something that is not there.
+  if (mutt_str_strcasestr(str, target) == NULL)
+    return 1;
```

Comment:
Do we want to do this (expensive) check unnecessarily?
We might as well let the `while` do it.